tests/test_runners/test_azure_runner.py

```diff
@@ -183,6 +183,34 @@ def test_show_terraform_outputs(
             assert output in captured.out
 
 
+def test_provision(matcha_testing_directory: str, template_runner: AzureRunner):
+    """Test service can provision resources using terraform.
+
+    Args:
+        matcha_testing_directory (str): Testing directory
+        template_runner (AzureRunner): a AzureRunner object instance
+    """
+    os.chdir(matcha_testing_directory)
```

Comment:
I think within the `matcha_testing_directory` mock we change the directory so this may be redundant, however, please check if this is the case if it is removed.

Reply:
We `chdir` back to the original wd once the testing directory is yielded.